lib/generator.js

```diff
@@ -283,20 +291,27 @@
         }
         
         if (deferred.promise.isPending()) {
-            readFile(resolve(__dirname, path), {encoding: "utf8"}, function (err, data) {
-                var id;
-
-                if (err) {
-                    deferred.reject(err);
-                } else {
-                    data = "var params = " + paramsString + ";\n" + data;
-                    id = self._photoshop.sendCommand(data);
-                    self._messageDeferreds[id] = deferred;
-
-                    deferred.promise.finally(function () {
-                        delete self._messageDeferreds[id];
-                    });
-                }
+            var id;
+            if (self._sendJSXCache.hasOwnProperty(path) && self._sendJSXCache[path] !== "") {
+                var data = "var params = " + paramsString + ";\n" + data;
+                id = self._photoshop.sendCommand(data);
+                self._messageDeferreds[id] = deferred;
+            } else {
+                readFile(resolve(__dirname, path), {encoding: "utf8"}, function (err, data) {
+                    if (err) {
+                        deferred.reject(err);
+                        return deferred;
+                    } else {
+                        self._sendJSXCache[path] = data;
+                        data = "var params = " + paramsString + ";\n" + data;
+                        id = self._photoshop.sendCommand(data);
+                        self._messageDeferreds[id] = deferred;
```

Comment:
In a perfect world, it would be nice if we didn't have this duplicated code. It would also be nice if this method were rewritten to a.) use a promisified version of readFile so the code were cleaner, and b.) leverage the `_sendJSXString` method above.

But, this isn't a perfect world. I'll spend a few minutes seeing if I can clean this up, or give up.

Also, I agree that it seems bad to hit the FS every time we execute JSX, but really, the OS should do a good job of caching this. It would be interesting to know what the real world performance improvement is.